R/build-home-authors.R

```diff
@@ -105,22 +105,56 @@ author_list <- function(x, authors_info, comment = FALSE) {
   roles <- paste0(role_lookup[x$role], collapse = ", ")
   substr(roles, 1, 1) <- toupper(substr(roles, 1, 1))
 
+  orcid <- x$comment %>% purrr::pluck('ORCID')
+  x <- remove_orcid_comment(x)
+
   list(
     name = name,
     roles = roles,
-    comment = x$comment
+    comment = x$comment,
+    orcid = orcid_link(orcid)
   )
 }
 
 author_desc <- function(x, comment = TRUE) {
   paste(
     x$name,
     "<br />\n<small class = 'roles'>", x$roles, "</small>",
+    if (!is.null(x$orcid))
+      x$orcid,
     if (comment && !is.null(x$comment))
       paste0("<br/>\n<small>(", x$comment, ")</small>")
   )
 }
 
+remove_orcid_comment <- function(x) {
```

Comment:
I think this would be simpler with another building block:

```R
remove_name <- function(x, name) {
  stopifnot(is.character(name), length(name) == 1)
  
  nms <- names(x)
  if (is.null(nms)) {
    return(x)
  }
  
  x[!(nms %in% name)]
}

p0 <- person("H", "W")
p1 <- person("H", "W", comment = "one")
p2 <- person("H", "W", comment = c("one", "two"))
p3 <- person("H", "W", comment = c("one", ORCID = "orcid"))
p4 <- person("H", "W", comment = c(ORCID = "orcid"))
p5 <- person("H", "W", comment = c(ORCID = "orcid1", ORCID = "orchid2"))

remove_name(p0$comment, "ORCID")
remove_name(p1$comment, "ORCID")
remove_name(p2$comment, "ORCID")
remove_name(p3$comment, "ORCID")
remove_name(p4$comment, "ORCID")
remove_name(p5$comment, "ORCID")
```

And would you mind turning those into unit tests?